Add THP_API to THPGenerator_Wrap

pbelevich opened the pull request:
Fixes:
```
error LNK2001: unresolved external symbol "struct _object * __cdecl THPGenerator_Wrap(struct at::Generator)" (?THPGenerator_Wrap@@YAPEAU_object@@UGenerator@at@@@Z)
build\lib.win-amd64-3.6\torch_test_cpp_extension\rng.cp36-win_amd64.pyd : fatal error LNK1120: 1 unresolved externals
```

I forgot it, my fault

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35167 Move normal() to DistributionTemplates
* #35199 Fix  _copysign is not a member of std (Windows)
* **#35194 Add THP_API to THPGenerator_Wrap**

Differential Revision: [D20591604](https://our.internmc.facebook.com/intern/diff/D20591604)

pbelevich committed:
Add THP_API to THPGenerator_Wrap

[ghstack-poisoned]

pbelevich committed:
Update on "Add THP_API to THPGenerator_Wrap"

Fixes:
```
error LNK2001: unresolved external symbol "struct _object * __cdecl THPGenerator_Wrap(struct at::Generator)" (?THPGenerator_Wrap@@YAPEAU_object@@UGenerator@at@@@Z)
build\lib.win-amd64-3.6\torch_test_cpp_extension\rng.cp36-win_amd64.pyd : fatal error LNK1120: 1 unresolved externals
```




[ghstack-poisoned]

pbelevich committed:
Update on "Add THP_API to THPGenerator_Wrap"

Fixes:
```
error LNK2001: unresolved external symbol "struct _object * __cdecl THPGenerator_Wrap(struct at::Generator)" (?THPGenerator_Wrap@@YAPEAU_object@@UGenerator@at@@@Z)
build\lib.win-amd64-3.6\torch_test_cpp_extension\rng.cp36-win_amd64.pyd : fatal error LNK1120: 1 unresolved externals
```




[ghstack-poisoned]

pbelevich committed:
Update on "Add THP_API to THPGenerator_Wrap"

Fixes:
```
error LNK2001: unresolved external symbol "struct _object * __cdecl THPGenerator_Wrap(struct at::Generator)" (?THPGenerator_Wrap@@YAPEAU_object@@UGenerator@at@@@Z)
build\lib.win-amd64-3.6\torch_test_cpp_extension\rng.cp36-win_amd64.pyd : fatal error LNK1120: 1 unresolved externals
```




[ghstack-poisoned]

peterjc123 commented on aten/src/ATen/native/Math.h:
```diff
@@ -79,7 +80,11 @@ Date:  February 1996
   T y_abs = std::abs(y);
   if(y_abs > 1.0) return std::numeric_limits<T>::quiet_NaN();
+#ifdef _WIN32
+  if(y_abs == 1.0) return copysign(std::numeric_limits<T>::infinity(), y);
```
Why is this one needed?

pbelevich committed:
Update on "Add THP_API to THPGenerator_Wrap"

Fixes:
```
error LNK2001: unresolved external symbol "struct _object * __cdecl THPGenerator_Wrap(struct at::Generator)" (?THPGenerator_Wrap@@YAPEAU_object@@UGenerator@at@@@Z)
build\lib.win-amd64-3.6\torch_test_cpp_extension\rng.cp36-win_amd64.pyd : fatal error LNK1120: 1 unresolved externals
```




[ghstack-poisoned]
